pandas/core/generic.py

```diff
@@ -1303,8 +1303,59 @@ def __invert__(self):
 
     def equals(self, other):
         """
-        Determines if two NDFrame objects contain the same elements. NaNs in
-        the same location are considered equal.
+        Test whether two DataFrame objects contain the same elements.
+
+        This function allows two DataFrame objects to be compared against
+        each other to see if they same shape and elements. NaNs in the same
+        location are considered equal. The column headers do not need to have
+        the same type, but the elements within the columns must be
+        the same dtype.
+
+        Parameters
+        ----------
+        other : DataFrame
+            The other DataFrame to be compared with the first.
+
+        Returns
+        -------
+        bool
+            True if all elements are the same in both DataFrames, False
+            otherwise.
+
+        See Also
+        --------
+        pandas.Series.eq : Compare two Series objects of the same length
+            and return a Series where each element is True if the element
+            in each Series is equal, False otherwise.
+
+        numpy.array_equal : Return True if two arrays have the same shape
+            and elements, False otherwise.
+
+        Notes
+        -----
+        This function requires that the elements have the same dtype as their
+        respective elements in the other DataFrame. However, the indices do
+        not need to have the same type, as long as they are still considered
+        equal.
+
+        Examples
+        --------
+        >>> a = pd.DataFrame({1:[0], 0:[1]})
+        >>> b = pd.DataFrame({1.0:[0], 0.0:[1]})
```

Comment:
to be consistent with other docstrings, it'd be good to use `df` for the main DataFrame the ones in which we'll call the method `df.equals()`, and something more descriptive for the other (e.g. `exactly_equal`, `different_data_type`, `different_index_type`...)

After creating the data, it'd be nice to display the content (i.e. `>>> df`)

Not sure in this case if it'd be better or worse, but in general we try to have somehow real-world (meaningful) data for the examples.